From d00b521f26eef3033f40ae54534eb721b2cbd456 Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jonh.wendell@redhat.com>
Date: Tue, 8 Feb 2022 18:10:13 -0500
Subject: Only setup the credentials if there is one (#37219)

Otherwise the credential helpers will be installed globally and this
might mess with pulls from `gcr.io`.
---
 prow/lib.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/prow/lib.sh b/prow/lib.sh
index e3b4a873d8..cfea59976b 100755
--- a/prow/lib.sh
+++ b/prow/lib.sh
@@ -41,6 +41,11 @@ function trace() {
 }
 
 function setup_gcloud_credentials() {
+  # Only setup the credentials if there is one
+  if [[ -z "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+    return
+  fi
+
   if [[ $(command -v gcloud) ]]; then
     gcloud auth configure-docker -q
   elif [[ $(command -v docker-credential-gcr) ]]; then
-- 
2.35.3

